File: data/src/main/java/home/oleg/placesnearme/database/AppDatabase.java
package home.oleg.placesnearme.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import home.oleg.placesnearme.dao.DetailedVenueWithPhotosDao;
import home.oleg.placesnearme.model.DetailedVenueDbEntity;
import home.oleg.placesnearme.model.PhotoDbEntity;

@Database(entities = {DetailedVenueDbEntity.class, PhotoDbEntity.class},
        version = 1)
public abstract class AppDatabase extends RoomDatabase implements DaoProvider {
    public static final String NAME = "places_near_me_db";

    public void clear() {
        getDetailedVenueWithPhotosDao().deleteAll();
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/database/AppDatabase.kt
package home.oleg.placesnearme.coredata.database

import androidx.room.Database
import androidx.room.RoomDatabase
import home.oleg.placesnearme.coredata.model.PlaceEntity
import home.oleg.placesnearme.coredata.model.DetailedVenueHistoryDbEntity
import home.oleg.placesnearme.coredata.model.PhotoEntity

@Database(entities = [
    PlaceEntity::class,
    PhotoEntity::class,
    DetailedVenueHistoryDbEntity::class], version = 1)
abstract class AppDatabase : RoomDatabase(), DaoProvider {

    fun clear() {
        detailedVenueWithPhotosDao.deleteAll()
    }

    companion object {
        val NAME = "places_near_me_db"
    }

}
